modules/plottinggl/plotters/scatterplotgl.cpp

```diff
@@ -355,6 +437,80 @@ void ScatterPlotGL::renderAxis(const size2_t &dims) {
                              size2_t(lowerLeft.x, upperRight.y - padding));
 }
 
+void ScatterPlotGL::objectPicked(PickingEvent *p) {
+    auto idToDataFrameIndex = [this](uint32_t id) -> std::tuple<bool, uint32_t> {
+        if (!indexColumn_) {
+            return std::tuple<bool, uint32_t>{false, 0};
+        }
+        auto &indexCol = indexColumn_->getTypedBuffer()->getRAMRepresentation()->getDataContainer();
+        auto it = util::find(indexCol, static_cast<uint32_t>(id));
+        if (it != indexCol.end()) {
+            return std::tuple<bool, uint32_t>{true, *it};
+        } else {
+            return std::tuple<bool, uint32_t>{false, 0};
+        }
+    };
+
+    const uint32_t id = static_cast<uint32_t>(p->getPickedId());
+    auto rowIndex = idToDataFrameIndex(id);
+
+    if (properties_.hovering_.get()) {
+        if (p->getState() == PickingState::Started) {
+            hoveredIndices_.insert(id);
+            processor_->invalidate(InvalidationLevel::InvalidOutput);
+        } else if (p->getState() == PickingState::Finished) {
+            hoveredIndices_.erase(id);
+            processor_->invalidate(InvalidationLevel::InvalidOutput);
+        }
+    }
+
+    auto logRowData = [&]() {
+        if (std::get<0>(rowIndex) && xAxis_ && yAxis_) {
+            std::ostringstream ss;
+            ss << "Index: " << std::get<1>(rowIndex) << "\n"
+               << properties_.xAxis_.getTitle() << ": "
+               << xAxis_->getRepresentation<BufferRAM>()->getAsDouble(id) << "\n"
+               << properties_.yAxis_.getTitle() << ": "
+               << yAxis_->getRepresentation<BufferRAM>()->getAsDouble(id);
+            if (color_) {
+                ss << "\nColor Value: " << color_->getRepresentation<BufferRAM>()->getAsDouble(id);
+            }
+            if (radius_) {
+                ss << "\nRadius: " << radius_->getRepresentation<BufferRAM>()->getAsDouble(id);
+            }
+            LogWarn(ss.str());
+        }
+    };
+
+    if (p->getEvent()->hash() == MouseEvent::chash()) {
+        auto me = p->getEventAs<MouseEvent>();
+        if (me->button() == MouseButton::Left) {
+            if (me->state() == MouseState::Release) {
+                // print information on current element
+                logRowData();
```

Comment:
Comment out debug logging?

Reply:
Logging here is on purpose to get the information of the picked point like position. Wishful thinking: tooltip information on hover instead :)